tensorflow/compiler/xla/service/gpu/gemm_rewriter.cc

```diff
@@ -1170,14 +1171,19 @@ class GemmRewriterVisitor : public DfsHloRewriteVisitor {
     config.set_epilogue(GemmBackendConfig::BIAS);
     std::vector<HloInstruction *> operands(gemm->operands().begin(),
                                            gemm->operands().end());
-    operands.push_back(bias);
-
+    if (gemm->custom_call_target() == kCublasLtMatmulF8CallTarget) {
+      // Matrix and vector bias don't co-exist for FP8 matmul.
+      config.set_beta(0.0);
```

Comment:
I think it's fine to limit the fusion to either a vector or a matrix bias for FP8 GEMMs. If beta isn't zero here, you're effectively overwriting an existing matrix bias though. FuseVectorBiasAdd probably shouldn't do anything if the Custom Call is FP8 and beta is not zero.

Reply:
Agreed we shouldn't overwrite the existing matrix bias.

Why can't the matrix and vector bias co-exist? Is this a cublas limitation? Even if cublas doesn't support this, I think C and the matrix bias should be separate operands, for consistency with the non-FP8 case.

Reply:
I don't think it's a cublasLT limitation, but it doesn't seem particularly relevant either. C currently contains the fused matrix bias when present.